src/fheroes2/dialog/dialog_adventure.cpp

```diff
@@ -21,106 +21,124 @@
-
-        if ( le.MouseClickLeft( buttonWorld.area() ) || Game::HotKeyPressEvent( Game::HotKeyEvent::WORLD_VIEW_WORLD ) ) {
-            result = Dialog::WORLD;
-            break;
-        }
-        if ( le.MouseClickLeft( buttonPuzzle.area() ) || Game::HotKeyPressEvent( Game::HotKeyEvent::WORLD_PUZZLE_MAP ) ) {
-            result = Dialog::PUZZLE;
-            break;
-        }
-        if ( le.MouseClickLeft( buttonInfo.area() ) || Game::HotKeyPressEvent( Game::HotKeyEvent::WORLD_SCENARIO_INFORMATION ) ) {
-            result = Dialog::INFO;
-            break;
-        }
-        if ( ( le.MouseClickLeft( buttonDig.area() ) || Game::HotKeyPressEvent( Game::HotKeyEvent::WORLD_DIG_ARTIFACT ) ) && buttonDig.isEnabled() ) {
-            result = Dialog::DIG;
-            break;
+    int selectAdventureOption( const bool enableDig )
+    {
+        // setup cursor
+        const CursorRestorer cursorRestorer( true, Cursor::POINTER );
+
+        fheroes2::Display & display = fheroes2::Display::instance();
+
+        const int32_t totalDialogWidth = 337;
+        const int32_t totalDialogHeight = 252;
+
+        // Prepare restorer of the adventure map when save feedback dialog is shown.
+        fheroes2::ImageRestorer back( display, ( display.width() - totalDialogWidth - BORDERWIDTH ) / 2, ( display.height() - totalDialogHeight + BORDERWIDTH ) / 2,
+                                      totalDialogWidth, totalDialogHeight );
+
+        fheroes2::StandardWindow background( 289, 204, true, display );
+
+        fheroes2::Button buttonWorld;
+        fheroes2::Button buttonPuzzle;
+        fheroes2::Button buttonInfo;
+        fheroes2::Button buttonDig;
+        fheroes2::Button buttonCancel;
+
+        const int32_t largeButtonsXOffset = 30;
+        const int32_t largeButtonsYOffset = 15;
+
+        const bool isEvilInterface = Settings::Get().isEvilInterfaceEnabled();
+        const int apanel = isEvilInterface ? ICN::APANELE : ICN::APANEL;
+
+        background.renderButton( buttonWorld, apanel, 0, 1, { largeButtonsXOffset, largeButtonsYOffset }, fheroes2::StandardWindow::Padding::TOP_LEFT );
+        background.renderButton( buttonPuzzle, apanel, 2, 3, { largeButtonsXOffset, largeButtonsYOffset }, fheroes2::StandardWindow::Padding::TOP_RIGHT );
+        background.renderButton( buttonInfo, isEvilInterface ? ICN::BUTTON_INFO_EVIL : ICN::BUTTON_INFO_GOOD, 0, 1, { largeButtonsXOffset, largeButtonsYOffset + 2 },
+                                 fheroes2::StandardWindow::Padding::CENTER_LEFT );
+        background.renderButton( buttonDig, apanel, 6, 7, { largeButtonsXOffset, largeButtonsYOffset + 2 }, fheroes2::StandardWindow::Padding::CENTER_RIGHT );
+        background.renderButton( buttonCancel, isEvilInterface ? ICN::BUTTON_SMALL_CANCEL_EVIL : ICN::BUTTON_SMALL_CANCEL_GOOD, 0, 1, { 0, 11 },
+                                 fheroes2::StandardWindow::Padding::BOTTOM_CENTER );
+
+        if ( !enableDig ) {
+            buttonDig.disable();
```

Comment:
To render the disabled button we should also call `.draw()` method here:
```suggestion
            buttonDig.disable();
            buttonDig.draw();
```